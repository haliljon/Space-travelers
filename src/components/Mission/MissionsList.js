import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { displayMissions, joinMission } from '../../redux/missions/missions';
import Mission from './Mission';

const MissionsList = () => {
  const dispatch = useDispatch();

  const missionsStore = useSelector((state) => state.missions);
  const { status, missions } = missionsStore;

  useEffect(() => {
    if (status === 'initial') {
      dispatch(displayMissions());
    }
  }, [dispatch, status]);

  return (
    <table className="table table-striped container border mt-5 mb-5">
      <tbody>
        <tr>
          <th>Mission</th>
          <th>Description</th>
          <th>Status</th>
          <th className="text-white">.</th>
        </tr>
        {missions.map((obj) => (
          <Mission
            key={obj.id}
            id={obj.id}
            joined={obj.joined}
            missionName={obj.name}
            description={obj.description}
            onClick={() => dispatch(joinMission(obj))}
          />
        ))}
      </tbody>
    </table>
  );
};

export default MissionsList;
