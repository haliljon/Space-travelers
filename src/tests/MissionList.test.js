import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import MissionsList from '../components/Mission/MissionsList';
import store from '../redux/configureStore';

it('matches snapshot', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <MissionsList />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
