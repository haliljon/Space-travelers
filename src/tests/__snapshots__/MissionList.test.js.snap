// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<table
  className="table table-striped container table-hover mt-5 mb-5"
>
  <tbody>
    <tr>
      <th>
        Mission
      </th>
      <th>
        Description
      </th>
      <th>
        Status
      </th>
      <th
        className="text-white"
      >
        .
      </th>
    </tr>
  </tbody>
</table>
`;
